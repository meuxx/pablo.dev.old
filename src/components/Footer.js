import React from 'react'

class Footer extends React.Component {
  render() {
    return (
      <div id="footer">
        <div className="inner">
          <ul className="icons">
            <li>
              <a
                href="mailto:contact:pablo.dev"
                className="icon fa-envelope-o"
                aria-label="Email"
              >
                <span className="label">Email</span>
              </a>
            </li>
            <li>
              <a
                href="https://gitlab.com/meuxx"
                className="icon fa-gitlab"
                rel="noopener noreferrer"
                target="_blank"
                aria-label="GitLab"
              >
                <span className="label">GitLab</span>
              </a>
            </li>
            <li>
              <a
                href="https://github.com/meuxx"
                className="icon fa-github"
                rel="noopener noreferrer"
                target="_blank"
                aria-label="GitHub"
              >
                <span className="label">GitHub</span>
              </a>
            </li>
          </ul>
        </div>
      </div>
    )
  }
}

export default Footer
