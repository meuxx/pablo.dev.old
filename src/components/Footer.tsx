import React from 'react'
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
import { useGoal } from 'gatsby-plugin-fathom'

import { SidebarFooter, SidebarIconWrapper, SidebarIcon, SidebarIconLink } from '../styles/styles'

const Footer: React.FC = () => {
  const handleMailGoal = useGoal('NMFGPZ35')
  const handleGitHubGoal = useGoal('TUDCIJWD')
  const handleLinkedInGoal = useGoal('VN1YOMWA')

  return (
    <SidebarFooter>
      <SidebarIconWrapper>
        <SidebarIcon>
          <SidebarIconLink
            href="mailto:contact@pablo.dev"
            rel="noopener noreferrer"
            target="_blank"
            aria-label="Email"
            title="Email"
            onClick={handleMailGoal}
          >
            <FontAwesomeIcon icon={['far', 'envelope']} />
            <span>Email</span>
          </SidebarIconLink>
        </SidebarIcon>
        <SidebarIcon>
          <SidebarIconLink
            href="https://github.com/meuxx"
            rel="noopener noreferrer"
            target="_blank"
            aria-label="GitHub"
            title="GitHub"
            onClick={handleGitHubGoal}
          >
            <FontAwesomeIcon icon={['fab', 'github']} />
            <span>GitHub</span>
          </SidebarIconLink>
        </SidebarIcon>
        <SidebarIcon>
          <SidebarIconLink
            href="https://www.linkedin.com/in/pablo-maceda"
            rel="noopener noreferrer"
            target="_blank"
            aria-label="LinkedIn"
            title="LinkedIn"
            onClick={handleLinkedInGoal}
          >
            <FontAwesomeIcon icon={['fab', 'linkedin']} />
            <span>LinkedIn</span>
          </SidebarIconLink>
        </SidebarIcon>
      </SidebarIconWrapper>
    </SidebarFooter>
  )
}

export default Footer
