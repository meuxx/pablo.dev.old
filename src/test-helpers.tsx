import { render } from '@testing-library/react'
import { Themed } from './styles/GlobalStyle'
import React from 'react'

const renderComponent: typeof render = (ui, options) => render(ui, { wrapper: Themed, ...options })

const mockComponent = (componentName: string): React.FC => (props) => {
  const { children, ...otherProps } = props
  return (
    <div data-component={componentName} data-props={JSON.stringify(otherProps, null, ' ')}>
      {children}
    </div>
  )
}

export { renderComponent, mockComponent }
