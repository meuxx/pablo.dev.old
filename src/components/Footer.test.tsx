import React from 'react'
import { renderComponent } from '../test-helpers'
import Footer from './Footer'

describe('Footer', () => {
  it('renders without crashing', () => {
    const { container } = renderComponent(<Footer />)

    expect(container).toBeDefined()
    expect(container).toMatchSnapshot()
  })
})
