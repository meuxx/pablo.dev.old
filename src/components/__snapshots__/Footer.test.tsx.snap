// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer renders without crashing 1`] = `
.c0 {
  bottom: auto;
  left: auto;
  position: relative;
  width: 100%;
}

.c1 {
  cursor: default;
  list-style: none;
  padding-left: 0;
  margin: 0;
}

.c2 {
  display: inline-block;
  padding: 0 1em 0 0;
}

.c2:last-child {
  padding-right: 0;
}

.c3 {
  -webkit-transition: color 0.2s ease-in-out;
  transition: color 0.2s ease-in-out;
  -webkit-text-decoration: none;
  text-decoration: none;
  font-size: 1.5em;
  border-bottom: none;
  position: relative;
  color: rgba(255,255,255,0.4);
}

.c3 > span {
  display: none;
}

.c3:hover {
  color: #ffffff;
}

@media screen and (max-width:980px) {
  .c0 {
    bottom: auto;
    left: auto;
    padding: 4em 4em 6em 4em;
    position: relative;
    text-align: center;
    width: 100%;
  }
}

@media screen and (max-width:980px) {
  .c1 {
    margin: 0 0 1em 0;
  }
}

<div>
  <div
    class="c0"
  >
    <ul
      class="c1"
    >
      <li
        class="c2"
      >
        <a
          aria-label="Email"
          class="c3"
          href="mailto:contact@pablo.dev"
          rel="noopener noreferrer"
          target="_blank"
          title="Email"
        >
          <svg
            class="far-envelope"
          />
          <span>
            Email
          </span>
        </a>
      </li>
      <li
        class="c2"
      >
        <a
          aria-label="GitHub"
          class="c3"
          href="https://github.com/meuxx"
          rel="noopener noreferrer"
          target="_blank"
          title="GitHub"
        >
          <svg
            class="fab-github"
          />
          <span>
            GitHub
          </span>
        </a>
      </li>
      <li
        class="c2"
      >
        <a
          aria-label="LinkedIn"
          class="c3"
          href="https://www.linkedin.com/in/pablo-maceda"
          rel="noopener noreferrer"
          target="_blank"
          title="LinkedIn"
        >
          <svg
            class="fab-linkedin"
          />
          <span>
            LinkedIn
          </span>
        </a>
      </li>
    </ul>
  </div>
</div>
`;
